lib/Core/Pagerfanta.php

```diff
@@ -377,12 +382,17 @@ private function notCachedNbResults(): bool
      */
     public function getNbPages()
     {
+
         $nbPages = $this->calculateNbPages();
 
         if (0 === $nbPages) {
             return $this->minimumNbPages();
         }
 
+        if ($this->maxNbPages !== 0 && $this->maxNbPages < $nbPages) {
```

Comment:
Hmm, I'm not entirely sure if I want 0 to act as a "magic number" of sorts meaning there isn't a maximum number.  How about making the new property nullable (as well as the argument for the setter) and replace 0 with null here.

Then the guard in the setter would reject a number less than 1.

Reply:
No problem for that.

Having an internal 0 permit to typehint the property.
```
private int $maxNbPages = 0;
```